runtime/src/block_component_processor.rs

```diff
@@ -133,6 +137,52 @@ impl BlockComponentProcessor {
         Ok(())
     }
 
+    fn enforce_alpenglow_clock_bounds(
+        bank: Arc<Bank>,
+        parent_bank: Arc<Bank>,
+        footer: &BlockFooterV1,
+    ) -> Result<(), BlockComponentProcessorError> {
+        // Get parent time from alpenglow clock account (nanosecond precision)
+        // If alpenglow clock hasn't been populated, don't enforce the bounds.
+        let Some(parent_time_nanos) = parent_bank.get_alpenglow_clock() else {
+            return Ok(());
+        };
+
+        let parent_slot = parent_bank.slot();
+        let current_time_nanos = footer.block_producer_time_nanos as i64;
+        let current_slot = bank.slot();
+
+        let (lower_bound_nanos, upper_bound_nanos) =
+            Self::alpenglow_time_bounds(parent_slot, parent_time_nanos, current_slot);
+
+        let is_valid =
+            lower_bound_nanos <= current_time_nanos && current_time_nanos <= upper_bound_nanos;
+
+        match is_valid {
+            true => Ok(()),
+            false => Err(BlockComponentProcessorError::AlpenglowClockOutOfBounds),
+        }
+    }
+
+    /// Given the parent slot, parent time, and working bank slot, calculate the lower and upper
+    /// bounds for the block producer time. We return (lower_bound, upper_bound), where both bounds
+    /// are inclusive. I.e., the working bank time is valid if
+    /// lower_bound <= working_bank_time <= upper_bound.
+    pub fn alpenglow_time_bounds(
+        parent_slot: Slot,
+        parent_time_nanos: i64,
+        working_bank_slot: Slot,
```

Comment:
nit: working_bank is typically used in leader context, would update this to just be slot